backend.pm

```diff
@@ -522,7 +540,9 @@ sub set_logfile
 sub check_cmd
 {
 	my ($self, $cmd) = @_;
-	return run_cmd($self, $cmd) != 127;
+	my $ret = run_cmd($self, $cmd);
+	return unless(defined($ret));
+	return $ret != 127;
```

Comment:
Have you actually managed to get undefined from the run_cmd here? I guess that it does not happen unless we set timeout, which we should do here anyway.

Reply:
ok, lets do with the next!